### ISSUE TITLE
Java 9+ SSL problems on Ubuntu 18.04

### ISSUE THREAD

**Issue by gothub** [NCEAS/metadig-engine#203]
Java 9+ programs can't access URLs via SSL on Ubuntu 18.04 (Bionic) with the certificate package (ca-certificates-java) that is installed. When a Java program tries to connect to a site via SSL, this exception is thrown:
```
java.lang.RuntimeException: Unexpected error: java.security.InvalidAlgorithmParameterException: the trustAnchors parameter must be non-empty
```

This problem is caused because of a mismatch between the certificate format Java expects and the format that is available in `/etc/ssl/certs/java/cacerts`. Java 8 uses JKS format, Java 9+ uses PKCS12, and Ubuntu 18.04 doesn't provide the proper format for new installations of Java, as explained here: https://bugs.launchpad.net/ubuntu/+source/ca-certificates-java/+bug/1739631. This bug report also details possible solutions.

The solution that is currently being used on the new k8s cluster is to downgrade to OpenJDK 8 and recreate the certs, as detailed in the Ubuntu bug report. A longer term solution would be to use a newer version of OpenJDK, upgrade to the new cert format, and update the SSL configuration to provide a password that is required by the newer cert format (also detailed in the bug report).

**Comment by mbjones** [comment]
It also appears that the fix has been released in Ubuntu 18.04.1, so upgradign our image to be based on that could fix the issue as well.  Note that the issue you cite also says there is an easier way to fix it than downngrading Java. Instead, you can keep the newer version of Java and do the following:

1. edit /etc/java-9-openjdk/security/java.security file. Find the line that says keystore.type = pkcs12 and change that to jks
2. rm /etc/ssl/certs/java/cacerts file
3. run "update-ca-certificates -f"

That will generate a new cacerts file in the old format that does not require a password.  This switch back to using jks as the keystore format seems to be what was implemented in 18.04.1.

**Comment by gothub** [comment]
Yep, that is an easier solution than converting to the new format.
